src/attributes/classes.js

```diff
@@ -56,20 +52,22 @@ jQuery.fn.extend({
 
 	removeClass: function( value ) {
 		var classes, elem, cur, curValue, clazz, j, finalValue,
-			proceed = arguments.length === 0 || typeof value === "string" && value,
-			i = 0,
-			len = this.length;
+			i = 0;
 
 		if ( jQuery.isFunction( value ) ) {
 			return this.each(function( j ) {
 				jQuery( this ).removeClass( value.call( this, j, getClass( this ) ) );
 			});
 		}
-		if ( proceed ) {
-			classes = ( value || "" ).match( rnotwhite ) || [];
 
-			for ( ; i < len; i++ ) {
-				elem = this[ i ];
+		if ( !arguments.length ) {
+			return this.attr( "class", "" );
```

Comment:
Good, we definitely want `.attr()` here since `.prop()` won't work on SVG elements. We should ensure we have a test with SVG if we don't already.